boot/seal.go

```diff
@@ -583,10 +624,18 @@ func recoveryBootChainsForSystems(systems []string, trbl bootloader.TrustedAsset
 				seedKernel, seedGadget = seedGadget, seedKernel
 			}
 
-			// get the command line
-			cmdline, err := composeCommandLine(currentEdition, ModeRecover, system, seedGadget.Path)
-			if err != nil {
-				return fmt.Errorf("cannot obtain recovery kernel command line: %v", err)
+			var cmdlines []string
+			modes, ok := modesForSystems[system]
+			if !ok {
+				return fmt.Errorf("internal error: no modes for system %q", system)
+			}
```

Comment:
This is just a personal preference: I'd move this block at the beginning of the `for` loop, and remove the check on `if len(modesForSystems) == 0 {`, which doesn't seem to be particularly useful.

Or if you think that that check can be useful, then it seems to me it could be moved outside of the `for` loop (though I'd just remove it).